CONN-4169 Add docs for new Bitwarden integration (#5107)
SumoLogic/sumologic-documentation: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/bitwarden.md

---
id: bitwarden-source
title: Bitwarden Source
sidebar_label: Bitwarden
tags:
  - cloud-to-cloud
  - bitwarden

> description: Learn how to collect the event log from the Bitwarden API.

Reviewer: ```suggestion
description: Learn how to collect event logs from the Bitwarden API.
```

---
import CodeBlock from '@theme/CodeBlock';
import ExampleJSON from '/files/c2c/bitwarden/example.json';
import MyComponentSource from '!!raw-loader!/files/c2c/bitwarden/example.json';
import TerraformExample from '!!raw-loader!/files/c2c/bitwarden/example.tf';
import useBaseUrl from '@docusaurus/useBaseUrl';

<img src={useBaseUrl('img/integrations/security-threat-detection/bitwarden.png')} alt="thumbnail icon" width="250"/>

> The Bitwarden source will collect the events logs from their API. The event log will return unique IDs for many of its fields which makes it difficult to understand the logs without the contextual names of those IDs. This integration will additionally query the group, member and policy API endpoints before every poll cycle so it has current lookup information for the event log IDs. The event log is modified to include additional fields from the lookups so data like group names, member names, member emails and policy types can be added.

Reviewer: ```suggestion
The Bitwarden source collects event logs from their API. Many fields in the event log contain unique IDs, making it difficult to interpret the logs without contextual names. To enhance readability, this integration queries the group, member, and policy API endpoints before each poll cycle, ensuring up-to-date lookup information for event log IDs. The event log is then enriched with additional fields, such as group names, member names, member emails, and policy types.
```

## Data collected

| API Endpoint       | Data                                                                                                                                                           |
|:-------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `/public/events`   | Ingests the events logs into Sumo Logic                                                                                                                        |

> | `/public/groups`   | Used to get the group names so we can include it in the event log data. The integration does not ingest logs directly from this endpoint.                      |
> | `/public/members`  | Used to get the member names and email addresses so we can include it in the event log data. The integration does not ingest logs directly from this endpoint. |
> | `/public/policies` | Used to get the policy type so we can include it in the event log data. The integration does not ingest logs directly from this endpoint.                      |

Reviewer: ```suggestion
| `/public/groups`   | Gets the group names so we can include it in the event log data. The integration does not ingest logs directly from this endpoint.                      |
| `/public/members`  | Gets the member names and email addresses so we can include it in the event log data. The integration does not ingest logs directly from this endpoint. |
| `/public/policies` | Gets the policy type so we can include it in the event log data. The integration does not ingest logs directly from this endpoint.                      |
```

## Setup

### Vendor configuration

**Prerequisites**

To collect event logs from the Bitwarden API, you must have a Bitwarden Enterprise account. The integration uses OAuth 2.0 Client Credentials. Follow the authentication [instructions in the authentication section of the public API page](https://bitwarden.com/help/public-api/). 

:::important

> If you are using a Self-Hosted installation, you will also need to fill out your OAuth Token URL. Please see [the details on the Bitwarden site](https://bitwarden.com/help/public-api/).

Reviewer: ```suggestion
If you are using a Self-Hosted installation, you must also provide your OAuth Token URL. For more details, see [Bitwarden's documentation](https://bitwarden.com/help/public-api/).
```
